frontend/src/Frontend/Examples/WebSocketChat/Main.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes        #-}
+{-# LANGUAGE RecursiveDo       #-}
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Frontend.Examples.WebSocketChat.Main where
+
+import           Data.Aeson         (encode,decode)
+import           Data.ByteString    as B
+import           Data.ByteString.Lazy (toStrict, fromStrict)
+import           Data.Monoid        ((<>))
+import qualified Data.Text          as T
+import           Data.Text (Text)
+import           GHCJS.DOM.HTMLElement       (focus)
+import           Language.Javascript.JSaddle
+import           Reflex
+import           Reflex.Dom         hiding (mainWidget)
+import           Reflex.Dom.Core    (mainWidget)
+import           Control.Monad.Fix (MonadFix)
+import           Control.Monad      (void)
+
+--------------------------------------------------------------------------------
+import           Common.Examples.WebSocketChat.Message
+--------------------------------------------------------------------------------
+
+main :: IO ()
+main = run $ mainWidget app
+
+--------------------------------------------------------------------------------
+
+-- TODO
+--  - factor out the performEvents (see keyboard -example)
+--  - factor out the message forming
+--  - factor out the textInput-button combos
+--  - add close connection button and associated message handling
+app
+  :: ( DomBuilder t m
+     , MonadFix m
+     , MonadHold t m
+     , PostBuild t m
+     , PerformEvent t m
+     , TriggerEvent t m
+     , Prerender js m
+     )
+  => m ()
+app = do
+  rec
+    msgEvDyn <- widgetHold loginWidget (messagingWidget <$ loggedInEv)
+    let
+      msgSendEv = switch (current msgEvDyn)
+      msgRecEv = fmapMaybe decodeOneMsg wsRespEv
+      eRecRespTxt = fmap showMsg msgRecEv
+      loggedInEv = fmapMaybe loginEv msgRecEv
+    wsRespEv <- prerender (return never) $ do
+      let sendEv = fmap ((:[]) . toStrict . encode) msgSendEv
+      ws <- webSocket "ws://localhost:8000/websocketchat" $ def & webSocketConfig_send .~ sendEv
```

Comment:
Is there a way to add the web socket backend URL from Route?

Reply:
One way would be to store the host information in `config/common/route` which could later be used with `obelisk-executable-config` in the `frontend` to retrieve the files present in `config` which the `backend` serves by default. https://github.com/obsidiansystems/obelisk/tree/master/lib/executable-config#ghcjs

Did an example over here https://github.com/obsidiansystems/obelisk/commit/f87aee775be38f931fe0b5bd3c02db3aa24dfe60 which brings up that `obelisk` should get some examples as well

Reply:
Right.. `obelisk` has many stuff which would need examples.

Reply:
But still this is not really complete, @luigy is it possible to get the `websocketchat` part also from the `BackendRoute` ?

Reply:
@dfordivam oh I missed that it was a backend url. @cgibbard is there a convenient way to render a backend url in a similar way we can do it for the frontend?

Reply:
Covered by 2ed8d24